### PR TITLE
Monitoring: Switch graphs from Plotly to Victory

### PR DESCRIPTION
**Pull request by kyoto**
- Modify `usePrometheusPoll` to also return API errors and to accept and
  optional `endTime` parameter.
- Modify `usePoll` to accept additional dependencies that trigger the
  poller to restart when they change.
- Make the link to the Prometheus UI optional.
- Remove the now unused `Line` graph class.

### CODE REVIEW

File: frontend/public/components/graphs/query-browser.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+import { Dropdown, humanizeNumber, LoadingInline, useRefWidth } from '../utils';
+import { formatPrometheusDuration, parsePrometheusDuration, twentyFourHourTime } from '../utils/datetime';
+import { PrometheusEndpoint } from './helpers';
+import { usePrometheusPoll } from './prometheus-poll-hook';
+import { PrometheusGraph } from './prometheus-graph';
+import { areaTheme } from './themes';
+
+const spans = ['5m', '15m', '30m', '1h', '2h', '6h', '12h', '1d', '2d', '1w', '2w'];
+const dropdownItems = _.zipObject(spans, spans);
```
Comment (spadgett) [inline frontend/public/components/graphs/query-browser.tsx]: nice, I learned a new lodash command :)

File: frontend/public/components/graphs/query-browser.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+};
+
+type GraphProps = {
+  colors: string[];
+  data: {
+    name: string;
+    values: GraphDataPoint[];
+  }[];
+  domain: Domain;
+  onZoom: (Domain) => void;
```
Comment (spadgett) [inline frontend/public/components/graphs/query-browser.tsx]: I believe this is naming the parameter `Domain`, not specifying the type.

File: frontend/public/components/graphs/query-browser.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  query: string;
+};
+
+type QueryBrowserProps = {
+  colors: string[];
+  GraphLink: React.ComponentType<any>;
+  metric: string;
+  onDataUpdate: (GraphDataMetric) => void;
```
Comment (spadgett) [inline frontend/public/components/graphs/query-browser.tsx]: Same comment here.

File: frontend/public/components/graphs/query-browser.tsx (hunk 2 above)
Comment (kyoto) [reply frontend/public/components/graphs/query-browser.tsx]: Oops. Fixed by #1614

File: frontend/public/components/graphs/query-browser.tsx (hunk 3 above)
Comment (kyoto) [reply frontend/public/components/graphs/query-browser.tsx]: 👍 Thanks

**Pull request merged**